editoast/editoast_authz/src/authorizer.rs

```diff
@@ -17,20 +18,44 @@ pub struct Authorizer<S: StorageDriver> {
 }
 
 impl<S: StorageDriver> Authorizer<S> {
+    /// Initialize an authorizer for the given user.
+    /// If the user doesn't exist, it will be created.
     #[tracing::instrument(skip_all, fields(%user), ret(level = Level::DEBUG), err)]
     pub async fn try_initialize(
         user: UserInfo,
         regulator: Regulator<S>,
     ) -> Result<Self, Error<S::Error>> {
-        let user_id = regulator
+        let user_info = regulator
             .driver
             .ensure_user(&user)
             .await
             .map_err(Error::Storage)?;
-        debug!(%user, %user_id, "user authenticated");
+        debug!(%user_info, "user authenticated");
         let authorizer = Self {
-            user,
-            user_id,
+            user: user_info.info,
+            user_id: user_info.id,
+            regulator,
+        };
+        Ok(authorizer)
+    }
+
+    /// Initialize an authorizer for the given user.
+    /// If the user doesn't exist, an error is returned.
+    #[tracing::instrument(skip_all, fields(%user), ret(level = Level::DEBUG), err)]
+    pub async fn initialize(
```

Comment:
Not using `try_` prefix when there is another function `try_initialize` looks weird since both functions returns `Result`. Maybe we should improve the naming, one with `try_initialize / try_impersonate`? Or `try_initialize_and_create / try_initialize`?